src/platform/tls_openssl.c

```diff
@@ -476,6 +476,10 @@ CxPlatTlsAddHandshakeDataCallback(
     CXPLAT_TLS_PROCESS_STATE* TlsState = TlsContext->State;
 
     QUIC_PACKET_KEY_TYPE KeyType = (QUIC_PACKET_KEY_TYPE)Level;
+    if (TlsContext->ResultFlags & CXPLAT_TLS_RESULT_ERROR) {
+        CXPLAT_DBG_ASSERT(CxPlatIsRandomMemoryFailureEnabled());
+        return -1;
+    }
```

Comment:
I don't quite understand this one. Can you explain it?

Reply:
When the server is processing the client initial crypto in `CxPlatTlsSetEncryptionSecretsCallback`, if the handshake key fails to allocate, the error flag is set and `-1` is returned to OpenSSL. However, OpenSSL still proceeds to call `CxPlatTlsAddHandshakeDataCallback` despite the failure which hits this assert `CXPLAT_DBG_ASSERT(KeyType == 0 || TlsState->WriteKeys[KeyType] != NULL);`.   The idea here is to be resilient to external library errors and not continue if an error has already occurred.